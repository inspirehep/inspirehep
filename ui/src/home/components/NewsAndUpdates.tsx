import React from 'react';
import { Row, Col, Card} from 'antd';
import useAxios from 'axios-hooks';
// @ts-ignore
import SanitizedHTML from 'react-sanitized-html';

import Loading from '../../common/components/Loading';
import ExternalLink from '../../common/components/ExternalLink';
import ContentBox from '../../common/components/ContentBox';
import { BLOG_URL } from '../../common/constants';

interface Post {
  id: string;
  link: string;
  title: {
    rendered: string;
  };
  excerpt: {
    rendered: string;
  };
  date: string;
}

const MOST_RECENT_BLOG_POSTS_URL = `${BLOG_URL}/wp-json/wp/v2/posts?per_page=3`;

function renderBlogPost(post: Post) {
  return (
    <ExternalLink href={post.link} key={post.id}>
      <Row justify="center">
        <Col sm={24} lg={14} className="mb2 __ContentBox__">
          <Card>
            <div className="pa2">
              <div className="flex flex-nowrap">
                <div className="flex-grow-1">
                  <span className="result-item-title">
                    <SanitizedHTML html={post.title.rendered} />
                  </span>
                </div>
              </div>
              <div className="mt1">
                <SanitizedHTML html={post.excerpt.rendered} />
              </div>
              <span className="db pt1 tr o-60">
                {new Date(post.date).toLocaleDateString()}
              </span>
            </div>
          </Card>
        </Col>
      </Row>
    </ExternalLink>
  );
}

const NewsAndUpdates = () => {
  const [{ data, loading }] = useAxios(MOST_RECENT_BLOG_POSTS_URL);

  return (
    <>
      <Row justify="center" className="__NewsAndUpdates__">
        <Col>
          <ContentBox className="container">
            {loading ? (
              <Loading />
            ) : data ? (
              data.map((post: Post) => (
                <div data-test-id="news-post">{renderBlogPost(post)}</div>
              ))
            ) : (
              <span>No new updates</span>
            )}
          </ContentBox>
        </Col>
      </Row>
      <Row justify="center">
        <Col>
          <ExternalLink
            className="db tc f5 mt4"
            href="https://blog.inspirehep.net/"
          >
            View all
          </ExternalLink>
        </Col>
      </Row>
    </>
  );
};

export default NewsAndUpdates;
