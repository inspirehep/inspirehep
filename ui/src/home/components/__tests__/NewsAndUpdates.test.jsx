import React from 'react';
import { shallow } from 'enzyme';

import NewsAndUpdates from '../NewsAndUpdates';

describe('NewsAndUpdates', () => {
  it('renders', () => {
    const wrapper = shallow(<NewsAndUpdates />);
    expect(wrapper).toMatchSnapshot();
  });
});
