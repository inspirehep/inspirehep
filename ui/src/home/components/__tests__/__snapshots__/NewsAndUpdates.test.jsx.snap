// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsAndUpdates renders 1`] = `
<Fragment>
  <Row
    className="__NewsAndUpdates__"
    justify="center"
  >
    <Col>
      <ContentBox
        className="container"
        leftActions={null}
        loading={false}
        rightActions={null}
        subTitle={null}
        title={null}
      >
        <Loading />
      </ContentBox>
    </Col>
  </Row>
  <Row
    justify="center"
  >
    <Col>
      <ExternalLink
        as="a"
        className="db tc f5 mt4"
        href="https://blog.inspirehep.net/"
      >
        View all
      </ExternalLink>
    </Col>
  </Row>
  <Row
    justify="center"
  >
    <Col
      className="mt5"
    >
      <Space
        size="middle"
      >
        Follow us on
        <LinkWithTargetBlank
          as="a"
          href="https://bsky.app/profile/inspirehep.net"
        >
          <AntdIcon
            component={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            style={
              Object {
                "color": "#1285FC",
                "fontSize": "25px",
              }
            }
          />
        </LinkWithTargetBlank>
        â€¢
        <LinkWithTargetBlank
          as="a"
          href="https://x.com/inspirehep"
        >
          <XOutlined
            style={
              Object {
                "color": "black",
                "fontSize": "24px",
              }
            }
          />
        </LinkWithTargetBlank>
      </Space>
    </Col>
  </Row>
</Fragment>
`;
